### DISCUSSION TITLE
Weird response_model behaviour

### DISCUSSION THREAD

**Discussion by marcosmmb** [fastapi/fastapi#7860] [Questions] [answered]
When trying to use a response_model to a object from my database there is an error, except if I access a property of the object.

```python
async def register_terminal(inp, user):
    new_terminal = await create_new_terminal(inp.name, inp.location_id)
    return {"terminal": new_terminal}
```

```python
@router.post("/register_terminal", response_model=RegisterOutput)
async def register_terminal_(*, inp: RegisterInput, user=get_user()):
    return await register_terminal(inp, user)
```

so this error is raised:

>     raise ValidationError(errors)
> pydantic.error_wrappers.ValidationError: 3 validation errors
> response -> terminal -> id
>   field required (type=value_error.missing)
> response -> terminal -> name
>   field required (type=value_error.missing)
> response -> terminal -> location_id
>   field required (type=value_error.missing)

but if I insert a line of code that will access an object property (new_terminal.id), the error won't appear:

```python
async def register_terminal(inp, user):
    new_terminal = await create_new_terminal(inp.name, inp.location_id)
    new_terminal.id
    return {"terminal": new_terminal}
```

**Comment by dmontagu** [comment]
In general, I would expect the issue is related to what is happening in the `BaseModel.from_orm` function.

I can't really tell what the issue is from just the output you are sharing here, and I recognize it might be hard to make a smaller self-contained example, but if you could I'd be happy to see if I can figure out why this is happening.

Alternatively, if you can run this function in a debugger and figure out precisely where there is a difference in behavior during the parsing, that might offer enough insight to figure out why this issue is happening and what could be done to fix it.

**Comment by tiangolo** [comment, marked as answer]
It looks a lot like you need to set `orm_mode = True`.

Check the SQL tutorial: https://fastapi.tiangolo.com/tutorial/sql-databases/

Otherwise, please create a self-contained example that replicates the issue so we can take a look.
